account_invoice_open_sale/models/account_invoice.py

```diff
@@ -0,0 +1,45 @@
+# coding: utf-8
+# © 2016 Opener B.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.translate import translate, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    has_sale_order = fields.Boolean(compute='_compute_has_sale_order')
+
+    @api.multi
+    def _compute_has_sale_order(self):
+        """ Determines if the button to show sale order will be shown on the
+        invoice """
+        if not self.ids:
+            return True
+        self.env.cr.execute("""
+        SELECT ail.invoice_id
+        FROM account_invoice_line AS ail
+        JOIN sale_order_line_invoice_rel AS rel ON ail.id = rel.invoice_line_id
+        WHERE ail.invoice_id IN %s
+        """, (tuple(self.ids),))
+        invoice_ids = self.env.cr.fetchall()
+
+        if not invoice_ids:
```

Comment:
It's dangerous not to assign values to the given records. It can lead to KeyErrors.